public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -80,20 +87,19 @@ function legacyCreateSpanLinkFactory(splitOpenFn: SplitOpen, traceToLogsOptions?
     // the moment. Issue is that the trace itself isn't clearly mapped to dataFrame (right now it's just a json blob
     // inside a single field) so the dataLinks as config of that dataFrame abstraction breaks down a bit and we do
     // it manually here instead of leaving it for the data source to supply the config.
-
-    const expr = getLokiQueryFromSpan(span, traceToLogsOptions);
```

Comment:
I think we should completely separate the logic for creating the Splunk and Loki queries here. Right now, they're very intertwined which will make it harder to maintain going forward. 

I think in `SpanLink()`, we should have a switch statement on the dataSourceSettings.type with a case for loki, a case for Splunk, and the default case returning undefined. Each case will call a function like `getLinkForLoki()` or `getLinkForSplunk()`. There will be a bit of duplication, but I think it'll be easier to work with.